### PR TITLE
Feat: Show input validation message when user's pay amount is 0

### PR DESCRIPTION
**Pull request by sticklo**
## What does this PR do and why?

- This PR shows an input validation message when user's pay amount is 0. 

https://github.com/jbx-protocol/juice-interface/issues/710

## Screenshots or screen recordings

image

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

**Comment by johnnyd-eth** [comment]
Hey @sticklo, nice pickup! Few comments:

- We don't want the message to show until someone actually clicks the pay button. We want the homepage to look **exactly** as is until the pay button is clicked and the amount is 0. 
- Let's make the font-size of the error message smaller: same size as 'Receive 150,994 tokens/1 ETH' 
- Think it might be worth trying putting the error message above the input, so we avoid the  'Receive 150,994 tokens/1 ETH' message jumping down when it appears. Try putting the error message above the input, and when it appears, let's try to keep the input and text below it in **the same position on the screen** (don't want elements jumping around)
- Also you've accidentally committed deleting the example.env file 😅

**Comment by aeolianeth** [comment]
@sticklo I would recommend looking into using the antd `Form.Item` component here: https://ant.design/components/form/#Form.Item

This way we might be able to leverage the existing validation text styling.

**Comment by sticklo** [comment]
thanks for the review guys @johnnyd-eth  @aeolianeth . Will work on them

**Comment by sticklo** [comment]
@aeolianeth  please take a look again

File: src/components/shared/inputs/Pay/PayInputGroup.tsx [hunk 1]
```diff
@@ -38,49 +38,58 @@ export default function PayInputGroup({
...
-          weight={weight}
-          tokenSymbol={tokenSymbol}
-          tokenAddress={tokenAddress}
-          weightingFn={weightingFn}
-        />
-      </div>
+    <>
+      {isErrorField && (
+        <span style={{ color: 'red', fontSize: '0.7rem' }}>
+          Pay amount must be greater than 0.
```
Comment (johnnyd-eth) [inline src/components/shared/inputs/Pay/PayInputGroup.tsx:52]: **Issue:** don't forget to mark any text you add to the UI for translations. Will need to wrap this in <Trans> tags. Then run `yarn i18n:extract`, and commit the new translation files in your PR

File: src/components/shared/inputs/Pay/PayInputGroup.tsx [hunk 2]
```diff
@@ -38,49 +38,58 @@ export default function PayInputGroup({
 
   const [payAmount, setPayAmount] = useState<string>('0')
   const [payInCurrency, setPayInCurrency] = useState<CurrencyOption>(ETH)
+  const [isErrorField, setIsErrorField] = useState<boolean>(false)
```
Comment (johnnyd-eth) [inline src/components/shared/inputs/Pay/PayInputGroup.tsx:41]: **Suggestion:** If we continue with the state route (not ant d form validation), I think we can get a slightly better state name: maybe 'errorMessageVisible'

**Comment by johnnyd-eth** [comment]
This is awesome @sticklo , love how you've got the message appearing when the user types. I also like that the message is on top now. 

I do agree with @aeolianeth though that it's worth leveraging Ant D field validation for this. It will provide some extra styling features like making the whole input go red, and will be more consistent with the other forms throughout the app.

To do this, you'll need to:
- wrap the input and pay button in <Form> component (look how <Form>s are used throughout the app)
- Pass `formItemProps` into the formattedNumberInput component with `rules` and a validator (see example below)

Screen Shot 2022-04-27 at 3 29 53 pm

- Will have to pass the formInstance into the payoutButton component and then run `await form.validateFields()` in the `onClick` inside the payoutButton here:

Screen Shot 2022-04-27 at 3 37 17 pm

- Then will have to do some CSS trickery to get the error message appearing above the above the input, ant d error messages appear below by default. Might be able to look into the Ant d docs to see if there's a nice way to get the error messages appearing above a field but I'm not sure 


This is a good challenge for you and I'm sure you can get it done 💪

However, @aeolianeth if you don't think it's worth @sticklo going through this effort then I think we should at least just get Sticklo's current error message showing when the button is clicked and the user hasn't typed anything yet

**Comment by sticklo** [comment]
@johnnyd-eth  thanks for the feedback, I would work on it.

**Comment by aeolianeth** [comment]
@sticklo @johnnyd-eth 

Thanks @johnnyd-eth for the great great review. Totally agree with all points mentioned.

I think that the implementation in this PR is a net-positive for the app though, and I think the current approach is okay for now.

There are a few improvements that we could make:
- source the `red` color from our theme
- make the warning appear if the user has never typed (for example, clicks the pay button immediately, without typing)
- wrap text in <Trans> to mark it for translation.

Here's a patch that implements the above suggestions. To apply the patch:
1. copy the diff below (you can use the copy to clipboard button on the top-right of the diff when you hover it)
1. in your local terminal, use `pbpaste | git apply` to apply the patch.

```
diff --git a/src/components/shared/inputs/Pay/PayInputGroup.tsx b/src/components/shared/inputs/Pay/PayInputGroup.tsx
index 87102f5f..f22ca769 100644
--- a/src/components/shared/inputs/Pay/PayInputGroup.tsx
+++ b/src/components/shared/inputs/Pay/PayInputGroup.tsx
@@ -9,11 +9,16 @@ import { CurrencyOption } from 'models/currencyOption'
 
 import { WeightFunction } from 'utils/math'
 
+import { ThemeContext } from 'contexts/themeContext'
+
+import { Trans } from '@lingui/macro'
+
 import PayInputSubText from './PayInputSubText'
 
 export type PayButtonProps = {
   payAmount: string
   payInCurrency: CurrencyOption
+  onError?: (error?: Error) => void
 }
 
 export default function PayInputGroup({
@@ -35,6 +40,9 @@ export default function PayInputGroup({
     currencyMetadata,
     currencies: { USD, ETH },
   } = useContext(CurrencyContext)
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
 
   const [payAmount, setPayAmount] = useState<string>('0')
   const [payInCurrency, setPayInCurrency] = useState<CurrencyOption>(ETH)
@@ -48,8 +56,8 @@ export default function PayInputGroup({
   return (
     <>
       {isErrorField && (
-        <span style={{ color: 'red', fontSize: '0.7rem' }}>
-          Pay amount must be greater than 0.
+        <span style={{ color: colors.text.failure, fontSize: '0.7rem' }}>
+          <Trans>Pay amount must be greater than 0.</Trans>
         </span>
       )}
       <div
@@ -87,7 +95,11 @@ export default function PayInputGroup({
         </div>
 
         <div style={{ textAlign: 'center', minWidth: 150 }}>
-          <PayButton payAmount={payAmount} payInCurrency={payInCurrency} />
+          <PayButton
+            payAmount={payAmount}
+            payInCurrency={payInCurrency}
+            onError={() => setIsErrorField(true)}
+          />
         </div>
       </div>
     </>
diff --git a/src/components/v1/V1Project/V1PayButton.tsx b/src/components/v1/V1Project/V1PayButton.tsx
index 462c8754..b4ec3658 100644
--- a/src/components/v1/V1Project/V1PayButton.tsx
+++ b/src/components/v1/V1Project/V1PayButton.tsx
@@ -8,10 +8,11 @@ import { formatWad, fromWad } from 'utils/formatNumber'
 import { decodeFundingCycleMetadata } from 'utils/v1/fundingCycle'
 import useWeiConverter from 'hooks/WeiConverter'
 import PayWarningModal from 'components/shared/PayWarningModal'
-import { CurrencyOption } from 'models/currencyOption'
 
 import { V1CurrencyOption } from 'models/v1/currencyOption'
 
+import { PayButtonProps } from 'components/shared/inputs/Pay/PayInputGroup'
+
 import { readNetwork } from 'constants/networks'
 import { disablePayOverrides } from 'constants/v1/overrides'
 import { V1_PROJECT_IDS } from 'constants/v1/projectIds'
@@ -22,10 +23,7 @@ import V1ConfirmPayOwnerModal from './modals/V1ConfirmPayOwnerModal'
 export default function V1PayButton({
   payAmount,
   payInCurrency,
-}: {
-  payAmount: string
-  payInCurrency: CurrencyOption // TODO make the V1CurrencyOption
-}) {
+}: PayButtonProps) {
   const { projectId, currentFC, metadata, isArchived, terminal } =
     useContext(V1ProjectContext)
 
diff --git a/src/components/v2/V2Project/V2PayButton/index.tsx b/src/components/v2/V2Project/V2PayButton/index.tsx
index 0de9c9bb..443e2a2f 100644
--- a/src/components/v2/V2Project/V2PayButton/index.tsx
+++ b/src/components/v2/V2Project/V2PayButton/index.tsx
@@ -10,19 +10,17 @@ import { V2_CURRENCY_USD } from 'utils/v2/currency'
 import PayWarningModal from 'components/shared/PayWarningModal'
 import useWeiConverter from 'hooks/WeiConverter'
 
-import { CurrencyOption } from 'models/currencyOption'
-
 import { V2CurrencyOption } from 'models/v2/currencyOption'
 
+import { PayButtonProps } from 'components/shared/inputs/Pay/PayInputGroup'
+
 import V2ConfirmPayModal from './V2ConfirmPayModal'
 
 export default function V2PayButton({
   payAmount,
   payInCurrency,
-}: {
-  payAmount: string
-  payInCurrency: CurrencyOption // TODO make the V2CurrencyOption
-}) {
+  onError,
+}: PayButtonProps) {
   const { projectMetadata, fundingCycleMetadata } = useContext(V2ProjectContext)
 
   const [payModalVisible, setPayModalVisible] = useState<boolean>(false)
@@ -60,7 +58,10 @@ export default function V2PayButton({
           style={{ width: '100%' }}
           type="primary"
           onClick={() => {
-            if (weiPayAmt?.eq(0)) return
+            if (weiPayAmt?.eq(0)) {
+              return onError?.()
+            }
+
             setPayWarningModalVisible(true)
           }}
           disabled={isPayDisabled}

```

**Comment by aeolianeth** [comment]
Hey @sticklo, were you still interesting in pursuing this PR?

**Comment by sticklo** [comment]
@aeolianeth Hii, yes I am, I'm sorry I had issues with my machine that  I have just now been able to sort out

**Comment by aeolianeth** [comment]
@sticklo thanks for making the changes!

**issue** let's display the message if the user tries to click the button without typing anything in

https://user-images.githubusercontent.com/94939382/167062697-e29a28ef-2045-4a5c-a050-2e874f3620aa.mp4

**Comment by sticklo** [comment]
@aeolianeth  please review boss

**Review by aeolianeth — APPROVED** [review]
@sticklo thanks so much for your persistence on this PR!

It looks really great - I'm happy to approve this ✅ . Nice work ser.

**Comment by sticklo** [comment]
@aeolianeth thank you very much for your patience on this

**Pull request merged**